symantecdlp_connector.py

```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
+        summary[DLP_JSON_TOTAL_INCIDENTS] = action_result.get_data_size()
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _validate_date(self, date_timestamp):
+        """ This function is used to validate date timestamp as per YYYY-MM-DD format or valid ISO 8601 format.
+
+        :param date_timestamp: Value of the date timestamp
+        :return: status(phantom.APP_SUCCESS/phantom.APP_ERROR)
+        """
+
+        regex = r'^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):' \
+                r'([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$'
+
+        match_iso8601 = re.compile(regex).match
+        try:
+            if match_iso8601(date_timestamp) is not None:
+                return phantom.APP_SUCCESS
+            elif datetime.strptime(date_timestamp, '%Y-%m-%d'):
+                return phantom.APP_SUCCESS
+        except Exception:
+            return phantom.APP_ERROR
+
+        return phantom.APP_ERROR
+
+    def _handle_get_incident(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # create zeep client
+        ret_val = self._create_client(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        config = self.get_config()
+        if (param.get(DLP_JSON_INCIDENT_LONG_ID) is None) and (param.get(DLP_JSON_INCIDENT_ID) is None):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                'Please provide value for atleast one of the parameter incident_long_id or incident_id.'
+            )
+
+        incident_long_id = self._validate_integers(action_result, param.get(DLP_JSON_INCIDENT_LONG_ID), DLP_JSON_INCIDENT_LONG_ID)
+
+        if not (param.get(DLP_JSON_INCIDENT_LONG_ID) is None) and (incident_long_id is None):
+            return action_result.get_status()
+
+        incident_id = self._validate_integers(action_result, param.get(DLP_JSON_INCIDENT_ID), DLP_JSON_INCIDENT_ID)
+
+        if not (param.get(DLP_JSON_INCIDENT_ID) is None) and (incident_id is None):
+            return action_result.get_status()
+
+        if not (incident_long_id is None) and (not (incident_id is None)):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                'Please provide only one of the value for incident_long_id or incident_id not both.'
+            )
+
+        include_violations = param.get(DLP_JSON_INCLUDE_VIOLATIONS)
+        include_history = param.get(DLP_JSON_INCLUDE_HISTORY)
+        ret_val, incident_detail = self._get_incident_details(
+            action_result, incident_id, incident_long_id, config[DLP_JSON_EXTRACT_COMPONENTS], include_violations, include_history
+        )
+
+        if (phantom.is_fail(ret_val)):
```

Comment:
Please remove the redundant parentheses.

Reply:
done